controllers/gce/utils/utils.go

```diff
@@ -123,13 +132,35 @@ func (n *Namer) SetClusterName(name string) {
 	n.clusterName = name
 }
 
+func (n *Namer) SetClusterFirewallName(firewall_name string) {
+	// Retain backwards compatible behavior where firewallName == clusterName.
+	n.nameLock.Lock()
+	defer n.nameLock.Unlock()
+	glog.Infof("Changing cluster %v firewall name to %v", n.clusterName, firewall_name)
+	n.firewallName = firewall_name
+}
+
 // GetClusterName returns the UID/name of this cluster.
 func (n *Namer) GetClusterName() string {
 	n.nameLock.Lock()
 	defer n.nameLock.Unlock()
 	return n.clusterName
 }
 
+// GetClusterNameAndFirewall returns the UID/name and firewall/name of this cluster.
+func (n *Namer) GetClusterAndFirewallNames() (string, string) {
+	n.nameLock.Lock()
+	defer n.nameLock.Unlock()
+	return n.clusterName, n.firewallName
+}
+
+// GetFirewallName returns the firewall name of this cluster.
+func (n *Namer) GetFirewallName() string {
+	n.nameLock.Lock()
+	defer n.nameLock.Unlock()
+	return n.firewallName
```

Comment:
Suggest defaulting this to clusterName if firewallName is unset, so we don't inadvertently break non-fed deployments by changing the name of the firewall and leaking it.